app/scripts/modules/core/pipeline/config/stages/executionWindows/ExecutionWindowDayPickerComponent.spec.js

```diff
@@ -0,0 +1,87 @@
+import dayPickerComponent from './ExecutionWindowDayPickerComponent';
+
+fdescribe('Component:  Execution Window Day Picker', () => {
+
+  const DAYS = [{
+    dow: 'sunday',
+    abbr: 'Sun'
+  }, {
+    dow: 'monday',
+    abbr: 'Mon'
+  }, {
+    dow: 'tuesday',
+    abbr: 'Tue'
+  }, {
+    dow: 'wednesday',
+    abbr: 'Wed'
+  }, {
+    dow: 'thursday',
+    abbr: 'Thu'
+  }, {
+    dow: 'friday',
+    abbr: 'Fri'
+  }, {
+    dow: 'saturday',
+    abbr: 'Sat'
+  }];
+
+  let $componentController;
+  beforeEach(window.module(dayPickerComponent));
+  beforeEach(window.inject((_$componentController_) => $componentController = _$componentController_));
+
+  it('should not have any days selected when the days property is not defined', () => {
+    const ctrl = $componentController('executionWindowDayPicker', null, {days: undefined});
+    expect(ctrl.days).toBeUndefined();
+  });
+
+  it('should not have any days selected when the days property is null', () => {
+    const ctrl = $componentController('executionWindowDayPicker', null, {days: null});
+    expect(ctrl.days).toBe(null);
+  });
+
+  it('should not have any days selected when the days property is an empty array', () => {
+    const ctrl = $componentController('executionWindowDayPicker', null, {days: []});
+    expect(ctrl.days.length).toBe(0);
+  });
+
+  DAYS.forEach((day, index) => {
+    it(`should have ${day.dow} selected when the days property contains ${index + 1}`, () => {
+      const ctrl = $componentController('executionWindowDayPicker', null, {days: [index + 1]});
+      expect(ctrl.days.length).toBe(1);
+      expect(ctrl.days[0]).toBe(index + 1);
+    });
+  });
+
+  it('should select all the days when the all button is clicked', () => {
+    const ctrl = $componentController('executionWindowDayPicker', null, {});
+    ctrl.all();
+    expect(ctrl.days.length).toBe(7);
+  });
+
+  it('should select none of the days when the none button is clicked', () => {
+    const ctrl = $componentController('executionWindowDayPicker', null, {});
+    ctrl.all();
+    expect(ctrl.days.length).toBe(7);
+    ctrl.none();
+    expect(ctrl.days.length).toBe(0);
+  });
+
+  it('should select just the weekdays when the weekday button is clicked', () => {
+    const ctrl = $componentController('executionWindowDayPicker', null, {});
+    ctrl.weekdays();
+    expect(ctrl.days.length).toBe(5);
+
+    const days = new Set(ctrl.days);
+    for (let i = 2; i < 7; i++) {
+      expect(days.has(i)).toBe(true);
+    }
```

Comment:
Aside from being cleaner, this would imply that order matters for the `days` property, which, on the client-side, it does not.  Can you confirm that days is order-dependent on the server side?  In fact, I'm wondering if this should be a `Set`... thoughts?